processor/src/main/java/com/linecorp/decaton/processor/runtime/internal/ProcessPipeline.java

```diff
@@ -152,6 +150,7 @@ Completion process(TaskRequest request, DecatonTask<T> task) throws InterruptedE
             processMetrics.tasksCompleteDuration.record(completeDuration);
 
             if (e != null) {
+                log.error("Uncaught exception thrown by processor {} for task {}", scope, request, e);
```

Comment:
Just a question: The reason to move logging here is?

Reply:
It is to gather exception processing code in one place & catch exceptionally completed future just in case.

>> I think propagate and logging un-caught exception thrown by async processor is nicer

> I guess this is not possible from Decaton-perspective because async-processing can be executed on arbitrary user-side threads.
Using deferred completion timeout-callback would help to identify the uncaught exception handling in async-processing: https://github.com/line/decaton/blob/master/processor/src/main/java/com/linecorp/decaton/processor/ProcessingContext.java#L119

Ah, I meant "logging un-caught exception" here as exceptionally finished future like below.

```
context.deferCompletion().completeWith(CompletableFuture.failedFuture(new RuntimeException("fail")));
```

Current default `completeWith` implementation [ignores exceptionally finished future](https://github.com/line/decaton/blob/d9543b907cb6af8090b4159eb7cd8fddb4cabac1/processor/src/main/java/com/linecorp/decaton/processor/DeferredCompletion.java#L36), but I feel it's kindful to log such error in the future. (Then, putting `log.error` here sounds more natural.)

Reply:
Thanks, I understood.